### PR TITLE
DEVDOCS-4771: [update] Tax Provider API, add item reference field

### PR DESCRIPTION
**Pull request by bc-andreadao**
# [DEVDOCS-4771]

[DEVDOCS-4771]: https://bigcommercecloud.atlassian.net/browse/DEVDOCS-4771?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by bc-andreadao** [comment]
@benpratt77 If you could send me a screenshot of an example request or add where `item_reference` goes in the schema (before or after `item_code`?), that would be appreciated

**Comment by bc-andreadao** [comment]
Planning on editing the examples if my schema is correct

File: reference/tax_provider.yml [hunk 1]
```diff
@@ -856,7 +856,10 @@ components:
           description: A unique identifier for this item used to map responses back to the corresponding item on the order.
         item_code:
           type: string
-          description: 'The UPC/SKU of the item. The UPC value will be used when both UPC and SKU values are available on the item.'
+          description: 'The UPC of the item. If UPC is not available, the SKU value will be used. Empty string if both UPC and SKU are not available.'
```
Comment (benpratt77) [inline reference/tax_provider.yml:859]: This is still UPC/SKU, the change itself is additive

File: reference/tax_provider.yml [hunk 2]
```diff
@@ -856,7 +856,10 @@ components:
...
+        item_reference:
+          type: string
+          description: 'The SKU of the item. Empty string if SKU is not available.'
```
Comment (benpratt77) [inline reference/tax_provider.yml:862]: 👍

File: reference/tax_provider.yml [hunk 3]
```diff
@@ -917,7 +920,10 @@ components:
           description: A unique identifier for this item used to map responses back to the corresponding item on the order.
         item_code:
           type: string
-          description: 'The UPC/SKU of the item. The UPC value will be used when both UPC and SKU values are available on the item.'
+          description: 'The UPC of the item. If UPC is not available, the SKU value will be used. Empty string if both UPC and SKU are not available.'
```
Comment (benpratt77) [inline reference/tax_provider.yml:923]: as per above

File: reference/tax_provider.yml (hunk 1 above)
Comment (bc-andreadao) [reply reference/tax_provider.yml]: Thanks, I edited the description

File: reference/tax_provider.yml (hunk 3 above)
Comment (bc-andreadao) [reply reference/tax_provider.yml]: Description edited 👍

**Review by benpratt77 — DISMISSED** [review]

File: reference/tax_provider.yml [hunk 4]
```diff
@@ -641,6 +653,7 @@ paths:
                         shipping:
                           id: string
                           item_code: string
+                          item_reference: '12345678'
```
Comment (benpratt77) [inline reference/tax_provider.yml:656]: 🍹 Is it worth keeping this in line with item_code and having `string`?
Also curious about the examples being a mix of 'string' and string types.

Comment (slsriehl) [reply reference/tax_provider.yml:656]: @benpratt77 these are automatically generated, for the most part.

**Review by benpratt77 — DISMISSED** [review]

**Review by slsriehl — APPROVED** [review]
nice work @bc-andreadao, I just added some quotes for typecasting purposes.

**Pull request merged**